src/types.rs

```diff
@@ -66,6 +66,8 @@ impl HttpClient {
             retries += 1;
             if retries >= RETRY_COUNT {
                 return Err(err);
+            } else {
```

Comment:
sleep from std?
I'm not sure if that's good idea in async code.

So in my case we had metadata server rejecting connections due sudden burst in which case multiple retries in short interval would be only even more harmful
I used 200ms just to make sure retries are done within 1 second

Reply:
> How long does a loop iteration take for you without timeout?

Current logic performs retries within 10 millisecond
I do not think it is good idea for metadata server because google most likely has some burst limits as we started encountering it recently with GKE's scheduled pods

Reply:
> sleep from std? I'm not sure if that's good idea in async code.

No, `Duration` from `std`.

> So in my case we had metadata server rejecting connections due sudden burst in which case multiple retries in short interval would be only even more harmful I used 200ms just to make sure retries are done within 1 second

Okay, how about we started with like 50ms and multiply by 2 on each iteration?

Reply:
Ok

Reply:
@djc I refactored logic
Note that 800ms interval will not be used because when retries == 5, function exists